src/Communicants/Communicant.cpp

```diff
@@ -235,6 +234,19 @@ void Communicant::commSend(const ZZ_p& num) {
     commSend(ustring(toSend, MOD_SIZE), MOD_SIZE);
 }
 
+void Communicant::commSend(const vec_ZZ& vec) {
+    Logger::gLog(Logger::COMM, "... attempting to send: vec_ZZ " + toStr(vec));
+
+    const ZZ base = power_ZZ(2, 256) + 1;
```

Comment:
why 2^256?  What if individual ZZ's are larger?  Please add a test that considers very large ZZ to catch this.